### PR TITLE
feat: Add a dictionary for all the letters with image, icon and tips in HomeScreen

### PR DESCRIPTION
**Pull request by SaFouNaldo**
## Added Images: 
- Uploaded all images from pic_a.jpg to pic_z.jpg in the res/drawable folder, allowing each letter to have a corresponding visual representation.

## Modular Tips in strings.xml: 
- Each letter now has its own descriptive tip stored in strings.xml (e.g., tip_a to tip_z). This approach improves modularity, allowing for easy updates or localization if needed.

## SignTipBox Composable: 
- Created a reusable SignTipBox composable that displays the details for each letter. SignTipBox takes the letter as a parameter and:

- Fetches the image and icon resources dynamically based on the letter.
- Displays the image for each letter with a consistent size of 200x200 pixels.
- Shows an icon and the associated tip text below the image for a clear and structured layout.
## Helper Functions for Resource Retrieval:

- getImageResId(letter: Char): Retrieves the image resource ID for each letter, dynamically using the letter as a reference.
- getIconResId(letter: Char): Retrieves the icon resource ID based on the letter.
- getTipResId(letter: Char): Retrieves the corresponding tip from strings.xml, ensuring each letter has a unique tip associated with it.

-This setup provides a streamlined and modular approach to displaying ASL letter information, making the dictionary feature efficient, scalable, and easy to maintain.

### CODE REVIEW

**Review by NBCCF025 — CHANGES_REQUESTED** [review]
This PR provides a robust and modular approach for integrating ASL letter information with images, tips, and icons in a clear and maintainable way. Here’s some feedback:

1. **Images in Drawable Folder**:
- Excellent choice in organizing all images under `res/drawable`.

2. **Modular Tips in `strings.xml`**:
- Storing letter tips in `strings.xml` is a good approach for modularity and future localization.

3. **Helper Functions**:
- Helper functions `getImageResId`, `getIconResId`, and `getTipResId` contribute to clean, readable code by abstracting resource retrieval. We could put them in the `Utils.kt` file in further changes.

## Summary:
I will just ask for the Following change before a merge : the image file `piv_v.jpg` should be removed from the resources.

Overall, this is a well-organized and scalable implementation that should support future enhancements. Let me know when it is done for an approval.

**Review by ThomasMPont — CHANGES_REQUESTED** [review]
## Additions

This is a pretty innocent new feature. I don't see it causing any problems.
Good job on compartmentalizing your changes, so that they don't affect existing code.

## Requested changes

I don't understand why the home screen has multiple dictionaries now. I think you should remove the old letter list, which your dictionary replaces. Perhaps you could recycle some of its logic to improve your own dictionary before deleting it.

I have requested a couple other changes in file comments, these are marked with this section's title.

## Tests

Testing even the smallest additions is essential. Nice work doing that.
Remember to check your test reports for edge cases you might have missed.

Testing smaller helper functions on top of more complex objects that use them, can help catch subtle mistakes in logic. Testing everything also helps developers find the origin of regression errors in the future.

## Files

Try to think of where each function you define belongs conceptually. A function only used in the home screen does not necessarily belong in HomeScreen.kt.

## Future changes

### Searching for letters

Personally, I don't find the experience of scrolling to a specific letter enjoyable.

A smart scroll bar with its own letter list, or a letter search feature, would help with finding a specific letter in the dictionary.

A floating "back to top" arrow could also save users some time.

### Indexing resources

As mentioned below, refactoring letter resources (e.g. gesture pictures) to be indexed in alphabetical order, could maybe replace the error prone `when` statements, that are repeated at multiple places in the codebase.
This could be made a future task.

### Layout

I'm not sure how I feel about the dictionary scrolling down this far. Maybe it could be moved into its own popup or subscreen. Using logic from the old letter list, you could have it scroll on its own, rather than having the user scroll so far away from the other home screen elements.

File: app/src/main/res/drawable/piv_v.jpg
[the remark this answers is not in the document]
Comment (SaFouNaldo) [reply app/src/main/res/drawable/piv_v.jpg:1]: It's done thanks for pointing it

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt [hunk 1]
```diff
@@ -199,3 +221,87 @@ fun ExerciseButton(exercise: Exercise, navigationActions: NavigationActions) {
...
+              .background(colorResource(R.color.white), RoundedCornerShape(8.dp))
+              .padding(8.dp)
+              .testTag("SignTipBox_$letter")) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.padding(8.dp)) {
+              // Displaying the main image, e.g., `pic_a.jpg`
+              Image(
+                  painter = painterResource(id = imageResId),
+                  contentDescription = "Image for letter $letter",
+                  modifier = Modifier.size(200.dp))
```
[the remark this answers is not in the document]
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:246]: A good suggestion I'll take that into account for another PR

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt [hunk 2]
```diff
@@ -199,3 +221,87 @@ fun ExerciseButton(exercise: Exercise, navigationActions: NavigationActions) {
         Text(exercise.name, modifier = Modifier.testTag("${exercise.name}ExerciseButtonText"))
       }
 }
+
+@SuppressLint("RememberReturnType")
```
[the remark this answers is not in the document]
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:225]: Done

File: app/src/main/res/drawable/pic_a.jpg
[the remark this answers is not in the document]
Comment (SaFouNaldo) [reply app/src/main/res/drawable/pic_a.jpg:1]: I didn't took the pictures all by myself. They were used from the dataset I trained my model in this [link](https://www.kaggle.com/datasets/grassknoted/asl-alphabet)

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt [hunk 3]
```diff
@@ -199,3 +221,87 @@ fun ExerciseButton(exercise: Exercise, navigationActions: NavigationActions) {
...
+                    text = tipText,
+                    fontSize = 16.sp,
+                    color = colorResource(R.color.black),
+                    modifier = Modifier.padding(8.dp))
+              }
+            }
+      }
+}
+
+private fun getImageResId(letter: Char): Int {
```
[the remark this answers is not in the document]
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:267]: Moved them

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt [hunk 4]
```diff
@@ -199,3 +221,87 @@ fun ExerciseButton(exercise: Exercise, navigationActions: NavigationActions) {
...
+  val resName = "pic_${letter.lowercaseChar()}"
+  return R.drawable::class.java.getDeclaredField(resName).getInt(null)
+}
+
+private fun getIconResId(letter: Char): Int {
+  val resName = "letter_${letter.lowercaseChar()}"
+  return R.drawable::class.java.getDeclaredField(resName).getInt(null)
+}
+
+private fun getTipResId(letter: Char): Int {
```
[the remark this answers is not in the document]
Comment (ThomasMPont) [reply app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:277]: I don't agree with this approach. Ideally, we would never have to clean up code on `main`. Unless you're running out of time, I strongly recommend doing the changes now.

**Comment by ThomasMPont** [comment]
> Hello, Thanks for your detailed review. The feature I added is simple :
> 
> * The dictionary contains images, icons and tips
> * The first "LetterDictionary" is there to help users to travel/navigate to the selected letter(That's what I'm implementing for next PR)
>   Everything will be clearer for my next PR

That makes sense. You may ignore those suggestions then.

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt]: Done @ThomasMPont look the last PR

**Review by ThomasMPont — APPROVED** [review]

**Comment by ThomasMPont** [comment]
Good initiative adding the new tests without being forced to!
